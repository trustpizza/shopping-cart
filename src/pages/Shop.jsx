import { useState, useEffect } from "react";
import ProductCard from "../components/ProductCard"; // Ensure the path is correct
import useProducts from "../services/useProducts"; // Assuming service file path

const Shop = () => {
  const { products, error, loading } = useProducts();

  return (
    <div className="bg-base-200 p-6">
      <h1 className="text-3xl font-bold text-center text-primary mb-6">Bananas for Sale</h1>
      
      <div className="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6 min-h-[200px] flex items-center justify-center">
        {loading && <div className="text-xl text-center col-span-full">Loading products...</div>}
        {error && <div className="text-xl text-center text-red-500 col-span-full">{error}</div>}
        {!loading && !error && products.map((product) => (
          <ProductCard 
            key={product.id} 
            title={product.title} 
            price={product.price} 
            image={product.image} 
            onClick={() => {}} 
          />
        ))}
      </div>
    </div>
  );
};


export default Shop;
