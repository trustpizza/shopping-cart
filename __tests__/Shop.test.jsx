import { describe, it, expect, beforeEach, vi } from "vitest";
import { render, screen } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import Shop from "../src/pages/Shop";

describe("Shop Component", () => {
  const mockProduct = {
    image: "https://via.placeholder.com/150",
    title: "Sample Product",
    price: "$19.99",
  };

  it("matches the snapshot", () => {
    const { asFragment } = render(
      <MemoryRouter initialEntries={['/shop']}>
        <Shop />
      </MemoryRouter>
    );;
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders correct heading", () => {
    render(
      <MemoryRouter initialEntries={['/shop']}>
        <Shop />
      </MemoryRouter>
    );
    const heading = screen.getByRole("heading");
    expect(heading.textContent).toMatch(/Bananas for Sale/i);
  });

  // it("shows the number of items in the cart", () => {
  //   // This test assumes the Shop page displays a Cart item count

  //   // Assuming you have a cart state or context displaying the number of items
  //   const cartCount = screen.getByText(/items in cart/i); // Update with correct text based on implementation
  //   expect(cartCount).toBeInTheDocument();
  //   expect(cartCount.textContent).toMatch(/\d+/); // Expect a number to be displayed
  // });
});
