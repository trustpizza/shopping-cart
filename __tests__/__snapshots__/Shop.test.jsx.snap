// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Shop Component > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="bg-base-200 p-6"
  >
    <h1
      class="text-3xl font-bold text-center text-primary mb-6"
    >
      Bananas for Sale
    </h1>
    <div
      class="grid grid-cols-1 sm:grid-cols-2 md:grid-cols-3 lg:grid-cols-4 gap-6 min-h-[200px] flex items-center justify-center"
    >
      <div
        class="text-xl text-center col-span-full"
      >
        Loading products...
      </div>
    </div>
  </div>
</DocumentFragment>
`;
